### DISCUSSION TITLE
Pls help to figure out how to force private DNS (nextDNS) while using WireGuard VPN in android app

### DISCUSSION THREAD

**Discussion by onetimecontributor** [celzero/rethink-app#1793] [Q&A] [answered]
I would like to use Private DNS instead of one from WireGuard VPN (protonVPN atm).
I need it to be in a form of DoT or DoH ([device_id]-[profile_id].dns.nextdns.io)

I'm sorry if I've either missed it in the user guide, or issues/discussions, or it's a trivial question.

I've tried to play with wireguard's "dns servers" setting, but couldn't quite complete it.
1. if I keep proton's vpn, obviously I see "this device is not using nextdns" (when I visit https://my.nextdns.io/)
2. if I keep it empty (auto) , then I get cloudflare 1.1.1.1
3. if I set nextDNS IPs (not a solution), all works fine and "All good!" (except that I can't specify/identify the device) ! but nextDNS doesn't recommend to use IPs for mobile clients and I don't want to introduce DDNS that is required for this feature to work.
4. if I set it to DoT [profile_id].dns.nextdns.io, I unexpectedly get "this device is using nextdns with no profile" and, for some reason, I see added trailing coma in the rdns "dns servers" field after I save the setting - ",xxx.dns.nextdns.io"
5. IPv6 obviously gives me "dns error" while trying to connect

Just in case, though it doesn't seem relevant #1271 , I've tried to set "never proxy dns".

**Comment by onetimecontributor** [comment]
anyone, please

**Comment by Tyxiel** [comment, marked as answer]
1. Use ADVANCED instead of SIMPLE wireguard. (No need to edit your wireguard config)
2. Select all APPs (If you want).
3. (Optional) Enable RESTRICT on your Wireguard Advanced Config for killswitch.
4. (Optional) Enabled ALWAYS-ON on your Wireguard Advanced Config to auto proxy new apps.
5. Set your NextDNS as DOT or DOH.

**Reply by onetimecontributor** [reply]
what's interesting, is that on a new setup (new rdns app, new imported vpn profile) it worked.

**Reply by onetimecontributor** [reply]
... while in the previous setup, when I've chosen "simple" while importing vpn config, I wasn't able to change DNS afterwards from "wireguard" to "custom" (via DNS tile on the main screen) no matter what I did (like switching to ADVANCED, etc). It was limited to "wireguard" dns (though I could change IP of the DNS) that limited me in use of nextdns features.
